packages/bratus-app/src/components/ComponentNode/ComponentNode.jsx

```diff
@@ -82,38 +85,65 @@ const ComponentNode = (node) => {
       : nodeName;
   };
 
-  const truncatedNodeName = truncateNodeName(node.data.label, nodeNameLength);
+  const content = (
+    <div>
+      <p>This component is used {node.data.component.timesUsed} times.</p>
+      <Divider />
+      <div style={{ display: 'flex', gap: '10px' }}>
+        <Button type="primary" href={`vscode://file/${node.data.path}`}>
+          Go To Code
+        </Button>
+        <Button
+          onClick={() => {
+            node.data.onShowNodeDetail(node);
+            setisPopoverVisible(false);
+          }}
+          type="primary"
+        >
+          View Code
+        </Button>
+      </div>
+    </div>
+  );
 
+  const truncatedNodeName = truncateNodeName(node.data.label, nodeNameLength);
   return (
-    <StyledNode
-      linesOfCode={node.data.linesOfCode}
-      componentBackground={componentBackground}
-      treeLayoutDirection={treeLayoutDirection}
-      isHighlighted={isHighlighted()}
-      bgColor={getBgColor}
-      fontColor={getFontColor()}
-      onDoubleClick={() => node.data.onShowNodeDetail(node)}
+    <Popover
+      visible={isPopoverVisible}
+      title={node.data.label}
+      trigger={'click'}
+      content={content}
+      onClick={() => setisPopoverVisible(true)}
```

Comment:
Would be nicer to have a setIsPopoverVisible(!isPopOverVisible), because we need more ways of closing the popover apart from the X button that you will add. So the interaction is smoother if the popover closes when u press the modal again.